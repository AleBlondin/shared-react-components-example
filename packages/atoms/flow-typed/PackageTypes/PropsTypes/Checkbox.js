// @flow

declare type CheckboxPropsType = {
  checked: ?boolean,
  children?: React$Node,
  className?: string,
  disabled?: boolean,
  name?: string,
  theme?: any,
  id?: string,
  onChange: EventType => void,
  showError?: boolean,
};
