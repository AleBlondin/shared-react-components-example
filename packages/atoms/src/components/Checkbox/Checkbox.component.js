// @flow

/*
 *  This component is exposed to be used outside of ComponentStudio, change it carefully
 */

import * as React from 'react';
import uniqueId from 'lodash/uniqueId';
import { ThemeProvider } from 'styled-components';
import DefaultStyleTheme from './Checkbox.theme';
import Style from './Checkbox.style';

const Checkbox = (props: CheckboxPropsType) => {
  const checkboxDefaultId = props.id || uniqueId('checkbox_');

  const additionnalProps = {};
  if (props.name) {
    additionnalProps.name = props.name;
  }

  return (
    // $FlowFixMe props.theme cannot be undefined because there is a default props
    <ThemeProvider theme={props.theme}>
      <Style.Container htmlFor={checkboxDefaultId} className={props.className}>
        <input
          type="checkbox"
          id={checkboxDefaultId}
          onChange={props.onChange}
          checked={props.checked}
          disabled={props.disabled}
          {...additionnalProps}
        />
        <Style.Checkbox
          checked={props.checked}
          disabled={props.disabled}
          showError={props.showError}
        />
        {props.children ? <Style.Label>{props.children}</Style.Label> : null}
      </Style.Container>
    </ThemeProvider>
  );
};

Checkbox.defaultProps = {
  disabled: false,
  className: '',
  id: null,
  onChange: () => {},
  theme: DefaultStyleTheme,
};

Checkbox.displayName = 'Checkbox';

export default Checkbox;
