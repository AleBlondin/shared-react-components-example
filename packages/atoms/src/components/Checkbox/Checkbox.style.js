// @flow

import styled from 'styled-components';

import { SharedComponentsColors, Grid } from 'shared-components-atoms/style';

import CheckIcon from './check-icon.svg';

const StyledCheckbox = styled.span`
  border: 2px solid
    ${props =>
      props.showError
        ? SharedComponentsColors.errors
        : SharedComponentsColors.greyBackground};
  background-color: ${props =>
    props.showError
      ? SharedComponentsColors.roseWhite
      : SharedComponentsColors.white};
  cursor: pointer;
  display: inline-block;
  height: ${Grid(3)};
  width: ${Grid(3)};
  flex-shrink: 0;

  ${props =>
    props.checked
      ? `
      background-color: ${props.theme.checkColor};
      background-image: url(${CheckIcon});
      background-position: center;
      background-repeat: no-repeat;
      background-size: 90%;
      border-color: ${props.theme.checkColor};
      `
      : ''}

  ${props =>
    props.disabled
      ? `
      background-color: #bdc3c7;
      border-color: #bdc3c7;
      cursor: default;
      `
      : ''}
`;

export default {
  Checkbox: StyledCheckbox,
  Container: styled.label`
    align-items: center;
    display: flex;
    cursor: pointer;

    // The IE 11 user-agent stylesheet styles input[type='checkbox'] which is
    // more specific than a class. Thus we need to use this rule to overwrite
    // the user agent stylesheet
    input[type='checkbox'] {
      display: none;
    }
  `,
  Label: styled.span`
    margin-left: ${Grid(2)};
    pointer-events: none;
  `,
};
