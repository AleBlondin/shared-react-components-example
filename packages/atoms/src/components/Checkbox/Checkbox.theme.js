// @flow

import { SharedComponentsColors } from 'shared-components-atoms/style';

export default {
  fontFamily: 'Lato',
  labelColor: `${SharedComponentsColors.greyBackground}`,
  checkColor: `${SharedComponentsColors.brandGreen}`,
};
