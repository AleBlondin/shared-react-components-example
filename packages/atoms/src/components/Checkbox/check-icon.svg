<?xml version="1.0" encoding="UTF-8"?>
<svg viewBox="0 0 10 7" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>check-icon</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Parcours-de-versement" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="3-Choix-de-mes-portefeuilles" fill-rule="nonzero" fill="#FFFFFF">
            <g id="Group-2">
                <g id="tick">
                    <path d="M9.25755037,0.15559216 C9.07446953,-0.0275069943 8.77764306,-0.0275069943 8.59452559,0.15559216 L2.97767792,5.77249477 L0.818653569,3.61347041 C0.635572729,3.43037126 0.338746255,3.43038957 0.155628787,3.61347041 C-0.0274703672,3.79655125 -0.0274703672,4.09337773 0.155628787,4.27647688 L2.64616553,6.766977 C2.82919143,6.95005784 3.12623767,6.94992964 3.30919031,6.766977 L9.25755037,0.818616942 C9.44064952,0.635536102 9.44063121,0.338691314 9.25755037,0.15559216 Z" id="Shape"></path>
                </g>
            </g>
        </g>
    </g>
</svg>