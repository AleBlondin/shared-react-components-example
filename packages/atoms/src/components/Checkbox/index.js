// @flow

export { default } from './Checkbox.component';
export { default as DefaultStyleTheme } from './Checkbox.theme';
