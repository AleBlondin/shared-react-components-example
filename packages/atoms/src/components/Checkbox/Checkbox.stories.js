import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { withKnobs, boolean, object } from '@storybook/addon-knobs';

import Checkbox from './Checkbox.component';
import Theme from './Checkbox.theme';

class CheckboxStory extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: true,
    };
    this.handleCheckBoxChange = this.handleCheckBoxChange.bind(this);
  }

  handleCheckBoxChange({ target }) {
    this.setState({
      checked: target.checked,
    });
  }

  render() {
    const { checked } = this.state;

    return (
      <Checkbox
        {...this.props}
        checked={checked}
        onChange={this.handleCheckBoxChange}
      />
    );
  }
}

storiesOf('Atoms', module)
  .addDecorator(withKnobs)
  .add('Checkbox', () => (
    <CheckboxStory
      disabled={boolean('disabled', false)}
      theme={object('Theme', Theme)}
      showError={boolean('showError', false)}
    >
      Label
    </CheckboxStory>
  ));
