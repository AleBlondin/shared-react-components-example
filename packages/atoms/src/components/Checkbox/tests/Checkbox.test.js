import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from '../Checkbox.component';

describe('<Checkbox />', () => {
  const onChange = jest.fn();

  it('should match snapshot with label', () => {
    const component = renderer.create(
      <Checkbox id="1" onChange={onChange} checked={false}>
        Label
      </Checkbox>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when checked', () => {
    const component = renderer.create(
      <Checkbox checked id="2" onChange={onChange} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when disabled', () => {
    const component = renderer.create(
      <Checkbox disabled checked={false} id="3" onChange={onChange} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when errored', () => {
    const component = renderer.create(
      <Checkbox checked={false} id="4" onChange={onChange} showError />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
