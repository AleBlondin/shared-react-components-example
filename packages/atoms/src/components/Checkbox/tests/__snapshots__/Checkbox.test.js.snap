// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should match snapshot when checked 1`] = `
.c1 {
  border: 2px solid #626262;
  background-color: #FFFFFF;
  cursor: pointer;
  display: inline-block;
  height: 15px;
  width: 15px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #00965E;
  background-image: url(IMAGE_MOCK);
  background-position: center;
  background-repeat: no-repeat;
  background-size: 90%;
  border-color: #00965E;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 input[type='checkbox'] {
  display: none;
}

<label
  className="c0"
  htmlFor="2"
>
  <input
    checked={true}
    disabled={false}
    id="2"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <span
    checked={true}
    className="c1"
    disabled={false}
  />
</label>
`;

exports[`<Checkbox /> should match snapshot when disabled 1`] = `
.c1 {
  border: 2px solid #626262;
  background-color: #FFFFFF;
  cursor: pointer;
  display: inline-block;
  height: 15px;
  width: 15px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #bdc3c7;
  border-color: #bdc3c7;
  cursor: default;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 input[type='checkbox'] {
  display: none;
}

<label
  className="c0"
  htmlFor="3"
>
  <input
    checked={false}
    disabled={true}
    id="3"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <span
    checked={false}
    className="c1"
    disabled={true}
  />
</label>
`;

exports[`<Checkbox /> should match snapshot when errored 1`] = `
.c1 {
  border: 2px solid #FF4348;
  background-color: #FFF9F9;
  cursor: pointer;
  display: inline-block;
  height: 15px;
  width: 15px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 input[type='checkbox'] {
  display: none;
}

<label
  className="c0"
  htmlFor="4"
>
  <input
    checked={false}
    disabled={false}
    id="4"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <span
    checked={false}
    className="c1"
    disabled={false}
  />
</label>
`;

exports[`<Checkbox /> should match snapshot with label 1`] = `
.c1 {
  border: 2px solid #626262;
  background-color: #FFFFFF;
  cursor: pointer;
  display: inline-block;
  height: 15px;
  width: 15px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 input[type='checkbox'] {
  display: none;
}

.c2 {
  margin-left: 10px;
  pointer-events: none;
}

<label
  className="c0"
  htmlFor="1"
>
  <input
    checked={false}
    disabled={false}
    id="1"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <span
    checked={false}
    className="c1"
    disabled={false}
  />
  <span
    className="c2"
  >
    Label
  </span>
</label>
`;
